org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/handlers/JDTLanguageServer.java

```diff
@@ -219,7 +219,25 @@ public JavaProtocolExtensions getJavaExtensions() {
 	@Override
 	public void didChangeConfiguration(DidChangeConfigurationParams params) {
 		logInfo(">> workspace/didChangeConfiguration");
+		long start = System.currentTimeMillis();
+		long timeout = start + 30000;
+		boolean wait = true;
```

Comment:
You should move JobHelper over src/main/java and use waitForJobs(IJobMatcher,maxWaitMillis)